src/hooks/useFetch.js

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+
+function useFetch(url) {
+    const [usersList, setUsersList] = useState([]);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState(null);
+
+    const fetchUsersList = async () => {
+        try {
+            setLoading(true);
+            const response = await fetch(url);
+            const data = await response.json();
+            setLoading(false);
+            setUsersList(data);
+
+        } catch (error) {
+            setError(error);
```

Comment:
There's a little bug here that might cause the app to load forever.
In case an error happens, you need to set the `loading` state to `false`.
If `fetch` on line 13 throws an error, it'll immediately go to the `catch` clause skipping the `setLoading` part.
You can always put `setLoading` outside of the `try/catch` to avoid repeating the code.

e.g:
```js
setLoading(true)
try {
  //code..
} catch (error) {
 //code 
}
setLoading(false)
```